lib/stream.js

```diff
@@ -173,14 +171,12 @@ export const getSpawnedResult = async ({
 		]);
 	} catch (error) {
 		spawned.kill();
-		return await Promise.all([
+		return Promise.all([
 			error,
 			waitForFailedProcess(processSpawnPromise, processErrorPromise, processExitPromise),
 			Promise.all(stdioPromises.map(stdioPromise => getBufferedData(stdioPromise, encoding))),
 			getBufferedData(allPromise, encoding),
 			Promise.allSettled(customStreamsEndPromises),
 		]);
-	} finally {
-		controller.abort();
```

Comment:
`finally` is not needed there. This function's `catch` block is guaranteed not to throw.
Moving the `controller` outside so that `lib/stream.js` only focuses on streams and process completion.